docs/modules/ROOT/pages/documentation/extensions/grpc/README.adoc

```diff
@@ -0,0 +1,415 @@
+
+[source, Java]
+----
+public class GrpcClient {
+    
+    private static final Logger LOGGER = Logger.getLogger(GrpcClient.class.getName());
+    private final PayaraServiceGrpc.PayaraServiceStub stub; //reference to the stub service implementation class
+    private CountDownLatch latch;
+    private AtomicReference<Throwable> error;
+
+    public static void main(String[] args) throws InterruptedException, MalformedURLException, URISyntaxException {
+        URL myURL = new URL("http://localhost:8080/fish.payara.samples.grpc.PayaraService"); // URL for the deployed gRPC service
+        final GrpcClient client = new GrpcClient(myURL); // creating client
+        client.communicate(); // call service
+    }
+
+    public GrpcClient(URL url) throws URISyntaxException {
+        final Channel channel = ManagedChannelBuilder.forAddress(url.getHost(), url.getPort())
+                .usePlaintext().build(); //creating channel to start communication to the service
+        this.stub = PayaraServiceGrpc.newStub(channel); //creating stub from the channel reference
+        this.error = new AtomicReference<>(null);
+    }
+
+    public void communicate() throws InterruptedException {
+        latch = new CountDownLatch(1); //this is to wait until the communication finish with the current thread
+        stub.communicate(request("Hello World"), new ResponseObserver()); //calling service and adding a ResponseObserver to process response
+        latch.await(20, TimeUnit.SECONDS); //timeout to wait response
+    }
+
+    public Throwable getError() {
+        return error.get();
+    }
+
+    private final class ResponseObserver implements StreamObserver<PayaraResp> {
+
+        @Override
+        public void onNext(PayaraResp response) { //to process the service response
+            LOGGER.log(Level.INFO, "Response received: \"{0}\".", response.getMessage()); // printing the response from the service
+        }
+
+        @Override
+        public void onError(Throwable t) { //method to process errors
+            LOGGER.log(Level.SEVERE, "Error received", t);
+            error.set(t);
+            latch.countDown();
+        }
+
+        @Override
+        public void onCompleted() {
+            latch.countDown(); //indicating that the communication complete for the current thread
+        }
+
+    }
+
+    private static final PayaraReq request(String message) {
+        return PayaraReq.newBuilder().setMessage(message).build(); //creating request with an String message
+    }
+}
+----
+
+[[execution-of-helloworld-application]]
+=== Execution of the "HelloWorld" application
+
+To execute the client application, build the project and run the following command:
+
+[source,shell]
+----
+${client app} mvn compile exec:java -Dexec.mainClass="fish.payara.samples.grpc.GrpcClient"
+----
+
+And you'll see the following log entries on the server that the service prints after processing the corresponding message: 
```

Comment:
And this error in server side:

```
[#|2022-03-11T10:48:52.625-0300|GRAVE|Payara 5.2021.9|javax.enterprise.system.core|_ThreadID=283;_ThreadName=admin-thread-pool::admin-listener(5);_TimeMillis=1647006532625;_LevelValue=1000;_MessageID=NCLS-CORE-00003;|
  Exception while running a command
java.lang.NullPointerException
        at org.glassfish.deployment.admin.GetDeploymentConfigurationsCommand.addToResultDDList(GetDeploymentConfigurationsCommand.java:148)
        at org.glassfish.deployment.admin.GetDeploymentConfigurationsCommand.execute(GetDeploymentConfigurationsCommand.java:136)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:556)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:552)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:360)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:551)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:582)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:574)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:360)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:573)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1497)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl.access$1300(CommandRunnerImpl.java:120)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1879)
        at com.sun.enterprise.v3.admin.CommandRunnerImpl$ExecutionContext.execute(CommandRunnerImpl.java:1755)
        at org.glassfish.admin.rest.utils.ResourceUtil.runCommand(ResourceUtil.java:272)
        at org.glassfish.admin.rest.utils.ResourceUtil.runCommand(ResourceUtil.java:240)
        at org.glassfish.admin.rest.resources.TemplateExecCommand.executeCommandLegacyFormat(TemplateExecCommand.java:156)
        at org.glassfish.admin.rest.resources.TemplateCommandGetResource.processGetLegacyFormat(TemplateCommandGetResource.java:76)
        at sun.reflect.GeneratedMethodAccessor192.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167)
        at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:475)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:397)
        at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
        at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:255)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
        at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
        at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234)
        at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680)
        at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:356)
        at org.glassfish.admin.rest.adapter.RestAdapter$2.service(RestAdapter.java:335)
        at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:189)
        at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:520)
        at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:217)
        at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:182)
        at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:156)
        at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:218)
        at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:95)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:260)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:177)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:109)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:88)
        at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:53)
        at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:524)
        at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:89)
        at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:94)
        at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:33)
        at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:114)
        at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:569)
        at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:549)
        at java.lang.Thread.run(Thread.java:748)
|#]

[#|2022-03-11T10:48:52.634-0300|GRAVE|Payara 5.2021.9|org.glassfish.admingui|_ThreadID=121;_ThreadName=admin-thread-pool::admin-listener(2);_TimeMillis=1647006532634;_LevelValue=1000;|
  RestResponse.getResponse() gives FAILURE.  endpoint = 'http://localhost:4848/management/domain/applications/application/payara-web/_get-deployment-configurations'; attrs = '{}'|#]

[#|2022-03-11T10:49:21.522-0300|INFORMAÃıES|Payara 5.2021.9|javax.enterprise.system.container.web.com.sun.web.security|_ThreadID=81;_ThreadName=http-thread-pool::http-listener-1(2);_TimeMillis=1647006561522;_LevelValue=800;|
  Context path from ServletContext:  differs from path from bundle: /|#]

[#|2022-03-11T10:49:21.610-0300|GRAVE|Payara 5.2021.9||_ThreadID=288;_ThreadName=glassfish-web-async-thread-3;_TimeMillis=1647006561610;_LevelValue=1000;|
  Exception in thread "glassfish-web-async-thread-3" |#]

[#|2022-03-11T10:49:21.616-0300|GRAVE|Payara 5.2021.9||_ThreadID=288;_ThreadName=glassfish-web-async-thread-3;_TimeMillis=1647006561616;_LevelValue=1000;|
  java.lang.IllegalStateException: Request already released from asynchronous mode
        at org.apache.catalina.connector.AsyncContextImpl.complete(AsyncContextImpl.java:287)
        at fish.payara.extension.grpc.servlet.ServletServerStream$Sink.lambda$cancel$2(ServletServerStream.java:302)
        at fish.payara.extension.grpc.servlet.ServletServerStream$ServletTransportState.runOnTransportThread(ServletServerStream.java:144)
        at fish.payara.extension.grpc.servlet.ServletServerStream$Sink.cancel(ServletServerStream.java:301)
        at io.grpc.internal.AbstractServerStream.cancel(AbstractServerStream.java:141)
        at fish.payara.extension.grpc.servlet.ServletAdapter$GrpcReadListener.onError(ServletAdapter.java:305)
        at org.apache.catalina.connector.InputBuffer$ReadHandlerImpl.processAllDataRead(InputBuffer.java:507)
        at org.apache.catalina.connector.InputBuffer$ReadHandlerImpl.access$800(InputBuffer.java:385)
        at org.apache.catalina.connector.InputBuffer$ReadHandlerImpl$2.run(InputBuffer.java:475)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
|#]
```

Reply:
Can you check if you have the last version of the community server?